src/mds/MDCache.cc

```diff
@@ -10051,6 +10061,29 @@ void MDCache::handle_snap_update(const cref_t<MMDSSnapUpdate> &m)
       do_realm_invalidate_and_update_notify(in, m->get_snap_op(), notify_clients);
     }
   }
+  mds->send_message(reply, m->get_connection());
+}
+
+void MDCache::handle_snap_update_reply(const cref_t<MMDSSnapUpdateReply> &m)
+{
+  mds_rank_t from = mds_rank_t(m->get_source().num());
+  dout(10) << __func__ << " " << *m << " from mds." << from << dendl;
+
+  if (mds->get_state() < MDSMap::STATE_RESOLVE &&
+      mds->get_want_state() != CEPH_MDS_STATE_RESOLVE) {
+    return;
+  }
+
+  CInode *in = get_inode(m->get_ino());
+  if (in) {
+    in->snap_update_ref--;
+    if (!in->snap_update_ref) {
```

Comment:
What happens when one of the snap_update_reply is not received because the mds might have gone down before replying ?

Reply:
It will wait infinitely here I think. Maybe we should set a timetout here ? Any better idea ?

Reply:
Maybe remove the mds from the set when _this_ MDS receives an mdsmap that has the peer mds missing?

Reply:
@lxbsz ping?

Reply:
Sorry, I missed this. Will try to fix it. Thanks @kotreshhr @vshankar

Reply:
> Maybe remove the mds from the set when _this_ MDS receives an mdsmap that has the peer mds missing?

This sounds good to me and appended a commit to fix this.